usr/share/rear/lib/uefi-functions.sh

```diff
@@ -39,46 +39,73 @@ function trim {
+    # If GRUB2_MODULES_LOAD is nonempty, it determines what modules to load
+    local modules=( ${GRUB2_MODULES_LOAD:+"${GRUB2_MODULES_LOAD[@]}"} )
 
     # Configuration file is optional for image creation.
-    [[ -n "$2" ]] && embedded_config="--config $2"
+    shift
+    if [[ -n "$1" ]] ; then
+        # graft point syntax. $1 will appear as /boot/grub/grub.cfg in the image
+        embedded_config="/boot/grub/grub.cfg=$1"
+        shift
+        # directories that should be accessible by GRUB2 (e.g. because they contain the kernel)
+        dirs=( ${@:+"$@"} )
+    fi
 
-    if has_binary grub-mkimage ; then
-        gmkimage=grub-mkimage
-    elif has_binary grub2-mkimage ; then
+    if has_binary grub-mkstandalone ; then
+        gmkstandalone=grub-mkstandalone
+    elif has_binary grub2-mkstandalone ; then
         # At least SUSE systems use 'grub2' prefixed names for GRUB2 programs:
-        gmkimage=grub2-mkimage
+        gmkstandalone=grub2-mkstandalone
     else
         # This build_bootx86_efi function is only called in output/ISO/Linux-i386/250_populate_efibootimg.sh
         # which runs only if UEFI is used so that we simply error out here if we cannot make a bootable EFI image of GRUB2
         # (normally a function should not exit out but return to its caller with a non-zero return code):
-        Error "Cannot make bootable EFI image of GRUB2 (neither grub-mkimage nor grub2-mkimage found)"
+        Error "Cannot make bootable EFI image of GRUB2 (neither grub-mkstandalone nor grub2-mkstandalone found)"
+    fi
+
+    # Determine what modules need to be loaded in order to access given directories
+    # (if the list of modules is not overriden by GRUB2_MODULES_LOAD)
+    if (( ${#dirs[@]} )) && ! (( ${#modules[@]} )) ; then
+        if has_binary grub-probe ; then
+            gprobe=grub-probe
+        elif has_binary grub2-probe ; then
+            # At least SUSE systems use 'grub2' prefixed names for GRUB2 programs:
+            gprobe=grub2-probe
+        else
+            LogWarn "Neither grub-probe nor grub2-probe found"
+            if test /usr/lib/grub*/x86_64-efi/partmap.lst ; then
+                LogWarn "including all partition modules"
+                modules=( $(cat /usr/lib/grub*/x86_64-efi/partmap.lst) )
+            else
+                Error "Can not determine partition modules, ${dirs[*]} would be likely inaccessible in GRUB2"
+            fi
+        fi
+
+        if [ -n "$gprobe" ]; then
+            # this is unfortunately only a crude approximation of the Grub internal probe_mods() function
+            modules=( $( for p in "${dirs[@]}" ; do
+                             $gprobe --target=fs "$p"
+                             $gprobe --target=partmap "$p" | sed -e 's/^/part_/'
+                             $gprobe --target=abstraction "$p"
+                         done | sort -u ) )
+        fi
+        Log "GRUB2 modules to load: ${modules:+${modules[*]}}"
```

Comment:
Could you move this log message outside its current `if` statement so one can always see what modules will be used?
Similarly to code you've removed: 
```
LogPrint "Using '$GRUB2_MODULES' modules to build $outfile"
```

Reply:
@gozora done, I also made it back into LogPrint to match what was there before.